notebooks/ndd_example.ipynb

```diff
@@ -0,0 +1,223 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "ed9dd0a2-6b12-436f-b6c3-02e0072115e3",
+   "metadata": {},
+   "source": [
+    "This notebook shows the results of query-time near-duplicate detection (NDD).  It shows the same query with and without the duplicates.  To use this notebook, first see [these instructions](https://sycamore.readthedocs.io/en/stable/welcome_to_sycamore/get_started.html) and start the Sycamore containers using `docker compose up`.  Then, ingest the college credit card marketing agreements data.  The documents come from [data.gov](https://catalog.data.gov/dataset/college-credit-card-marketing-agreements-data), but we have made them accessible via S3.  There are two ingestion commands to choose from, depending on how realistic or quick the demo should be:\n",
+    "\n",
+    "- All 1911 PDFs (~4 hours): `docker compose run sycamore_crawler_s3 aryn-public cccmad`\n",
+    "- 35 PDFs needed for demo (minutes): `docker compose run sycamore_crawler_s3 aryn-public cccmad-tiny`\n",
+    "\n",
+    "Note that for the full dataset, approximately 17 documents will fail to ingest for various reasons.  The Sycamore importer will keep retrying them, but the problems will persist.  It's not necessary for 100% of the documents to be ingested in order to run this example.  Once the importer queue shrinks to ~100 documents, it's OK to proceed.  Note that due to variations in ingestion, results may not exactly match what's described here."
+   ]
```

Comment:
I rewrote it slightly to add more newlines (made it much more readable to me). I'd like to break up step 3, but couldn't figure out how.  Results below:

```
This notebook shows the results of query-time near-duplicate detection (NDD).  It shows the same query with and without the duplicates.  

To use this notebook:
1. Follow [these instructions](https://sycamore.readthedocs.io/en/stable/welcome_to_sycamore/get_started.html) and start the Sycamore containers using `docker compose up`.
2. Ingest the college credit card marketing agreements data.  The documents come from [data.gov](https://catalog.data.gov/dataset/college-credit-card-marketing-agreements-data), but we have made them accessible via S3.  There are two ingestion commands to choose from, depending on how realistic or quick the demo should be:
    - All 1911 PDFs (~4 hours): `docker compose run sycamore_crawler_s3 aryn-public cccmad`
    - 35 PDFs needed for demo (minutes): `docker compose run sycamore_crawler_s3 aryn-public cccmad-tiny`
3. If you imported the full dataset, you can proceed once the importer queue shrinks to ~100 documents. It's not necessary for 100% of the documents to be ingested in order to run this example.   Due to variations in ingestion, results may not exactly match what's described here. Note that for the full dataset, approximately 17 documents will fail to ingest for various reasons.  The Sycamore importer will keep retrying them, but the problems will persist.
```